aws-java-sdk-core/src/main/java/com/amazonaws/retry/PredefinedRetryPolicies.java

```diff
@@ -217,7 +221,7 @@ public final long delayBeforeNextRetry(AmazonWebServiceRequest originalRequest,
                 scaleFactor = SCALE_FACTOR;
             }
 
-            long delay = (1 << retriesAttempted) * scaleFactor;
```

Comment:
it may not be important, but the delay actually is a long, and part of the overflow problem was that it was integer overflowing, not long overflowing.  Also, the perf of bitshifts and multiplies is totally insignificant relative to s3's perf, so that really shouldn't be a concern here.  If you're worried about optimizing perf, you should be looking elsewhere.